lib/blueprinter/v2.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+require 'blueprinter/v2/dsl'
+require 'blueprinter/v2/options'
+require 'blueprinter/v2/reflection'
+require 'blueprinter/v2/view_builder'
+
+module Blueprinter
+  # Base class for V2 Blueprints
+  class V2
```

Comment:
Reading through this again with fresh eyes, I'm less inclined to want to codify the version as part of the class itself, especially since it doesn't quite describe what the class _is_ as much. 

The versioning system we were originally going with does seem over-engineered based on our current expectations, but I'm curious if we could have a more expressive name, and then introduce those versioning semantics later on if needed (e.g. `class Blueprint` -> `Blueprint[3.0]`).

Otherwise, I'd lean more towards `BlueprintV2` if we want to stick with the explicit version in the name.

Reply:
I've made `V2` a module and moved this code to `V2::Base` for now.